pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -131,7 +132,12 @@ func NewManager(conf *Config) (Manager, lifecycle.PodAdmitHandler) {
 			Path: filepath.Join(conf.StateDirectory, localStorageStateFile),
 		},
 	}
-	klog.InfoS("Creating node shutdown manager",
+	if conf.Logger != nil {
```

Comment:
We can also decide to make Config.Logger a value and require all callers to set it. I think I already did that for all in-tree code.

Reply:
I was too cautious. Requiring `Config.Logger` to be a valid value is simpler and doable here because this is an internal API where we can and want to change all callers.

Updated...